common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -117,21 +117,32 @@ public static String genStreamChunkId(long streamId, int chunkId) {
 
   @Override
   public void connectionTerminated(Channel channel) {
+    boolean failedToReleaseBuffers = false;
+
     // Close all streams which have been associated with the channel.
     for (Map.Entry<Long, StreamState> entry: streams.entrySet()) {
       StreamState state = entry.getValue();
       if (state.associatedChannel == channel) {
         streams.remove(entry.getKey());
 
-        // Release all remaining buffers.
-        while (state.buffers.hasNext()) {
-          ManagedBuffer buffer = state.buffers.next();
-          if (buffer != null) {
-            buffer.release();
+        try {
+          // Release all remaining buffers.
+          while (state.buffers.hasNext()) {
+            ManagedBuffer buffer = state.buffers.next();
+            if (buffer != null) {
+              buffer.release();
+            }
           }
+        } catch (RuntimeException e) {
+          failedToReleaseBuffers = true;
+          logger.error("Exception trying to release remaining StreamState buffers", e);
         }
       }
     }
+
+    if (failedToReleaseBuffers) {
```

Comment:
instead of using a boolean flag, can we keep the actual exception (we can just keep the first one) and rethrow it? then we can retain the stack trace.